### PR TITLE
Fix gcc4.4 tests

### LINKED ISSUE #5454: httpcli and httpscli failure on gcc4.4 portability test
https://grpc-testing.appspot.com/job/gRPC_portability_master/481/language=c,scenario=linux_x64_gcc4.4/testReport/junit/(root)/tests/bins_dbg_httpcli_test/ and https://grpc-testing.appspot.com/job/gRPC_portability_master/481/language=c,scenario=linux_x64_gcc4.4/testReport/junit/(root)/tests/bins_dbg_httpscli_test/

```
D0226 22:01:27.240215380   17660 env_linux.c:77]             Warning: insecure environment read function 'getenv' used
D0226 22:01:27.240381929   17660 env_linux.c:77]             Warning: insecure environment read function 'getenv' used
E0226 22:01:27.240932752   17660 tcp_client_posix.c:173]     failed to connect to 'ipv6:[::1]:32767': socket error: connection refused
/usr/bin/env: python2.7: No such file or directory
D0226 22:01:32.242924042   17660 test_config.c:159]          test slowdown: machine=1.000000 build=1.000000 total=1.000000
D0226 22:01:32.243027323   17660 env_linux.c:77]             Warning: insecure environment read function 'getenv' used
I0226 22:01:32.243043037   17660 httpscli_test.c:79]         test_get
I0226 22:01:32.243051205   17660 httpscli_test.c:82]         requesting from localhost:1105
E0226 22:01:32.244421395   17719 tcp_client_posix.c:173]     failed to connect to 'ipv6:[::1]:1105': socket error: connection refused
E0226 22:01:32.244550439   17719 tcp_client_posix.c:173]     failed to connect to 'ipv4:127.0.0.1:1105': socket error: connection refused
E0226 22:01:32.244590640   17719 httpscli_test.c:63]         assertion failed: response

*******************************
Caught signal 6
bins/dbg/httpscli_test[0x579235]
/lib/libpthread.so.0(+0xeff0)[0x7f8cb2cb9ff0]
/lib/libc.so.6(gsignal+0x35)[0x7f8cb2970f15]
/lib/libc.so.6(abort+0x180)[0x7f8cb2973d20]
bins/dbg/httpscli_test[0x4043ca]
bins/dbg/httpscli_test[0x406202]
bins/dbg/httpscli_test[0x406731]
bins/dbg/httpscli_test[0x406687]
bins/dbg/httpscli_test[0x407ad0]
bins/dbg/httpscli_test[0x499104]
bins/dbg/httpscli_test[0x423e44]
/lib/libpthread.so.0(+0x68ca)[0x7f8cb2cb18ca]
/lib/libc.so.6(clone+0x6d)[0x7f8cb2a17abd]
```

### PR DESCRIPTION
**Pull request by jtattermusch**
Several fixes here:
-- The link `http://openssl.org/source/openssl-1.0.2f.tar.gz` is dead, which is causing https://github.com/grpc/grpc/issues/5821. For now, sticking with openssl 1.0.2f version to fix the flake on master quickly and prevent further problems.
-- it turns out debian squeeze it being killed (its LTS has ended end of Feb, and `apt-get update` is broken with vanilla debian:squeeze docker image). Switching to debian wheezy with explictly installed gcc4.4.
-- As a side effect, the docker image now has python 2.7 (which could in turn fix #5454)

### CODE REVIEW

**Comment by jtattermusch** [comment]
Building the cxx_squeeze_x64 docker image is till broken, something is odd with debian repos 

```
W: Failed to fetch http://httpredir.debian.org/debian/dists/squeeze-lts/Release  Unable to find expected entry  main/binary-amd64/Packages in Meta-index file (malformed Release file?)

W: Failed to fetch http://httpredir.debian.org/debian/dists/squeeze/main/binary-amd64/Packages.gz  404  Not Found [IP: 131.252.208.20 80]

W: Failed to fetch http://httpredir.debian.org/debian/dists/squeeze-updates/main/binary-amd64/Packages.gz  404  Not Found [IP: 131.252.208.20 80]

E: Some index files failed to download, they have been ignored, or old ones used instead.
```

**Comment by jtattermusch** [comment]
Turns out that `apt-get update` on a vanilla debian:squeeze docker image  fails these days.

**Comment by jtattermusch** [comment]
The new wheezy image has been successfully prebuilt on all workers: https://grpc-testing.appspot.com/view/All/job/gRPC_prebuild_dockerfiles/31/

**Comment by jtattermusch** [comment]
`tools/run_tests/run_tests.py -l c --use_docker --compiler gcc4.4` is not passing locally.

This should be ready for review.

FYI @ctiller @nicolasnoble

File: test/core/end2end/fixtures/h2_uchannel.c [hunk 1]
```diff
@@ -285,7 +285,7 @@ static void chttp2_init_client_micro_fullstack(grpc_end2end_test_fixture *f,
                                                grpc_channel_args *client_args) {
   micro_fullstack_fixture_data *ffd = f->fixture_data;
   grpc_connectivity_state conn_state;
-  grpc_connected_subchannel *connected;
+  grpc_connected_subchannel *is_connected;
```
Comment (murgatroid99) [inline test/core/end2end/fixtures/h2_uchannel.c]: Why make this change? This variable doesn't look like a boolean.

Comment (jtattermusch) [inline test/core/end2end/fixtures/h2_uchannel.c]: Ah, nice catch. This needed to be renamed from "connected" to some other name to fix a -Wshadow warning treated as error (there's some odd behavior under gcc4.4).

**Comment by jtattermusch** [comment]
PTAL.

**Comment by murgatroid99** [comment]
LGTM

**Comment by jtattermusch** [comment]
Merging as objC is not affected by this PR at all and this is fixing a breakage on portability master suite.

**Pull request merged**